bindings/java/src/org/sleuthkit/datamodel/Report.java

```diff
@@ -90,4 +115,186 @@ public String getSourceModuleName() {
+	}
+
+	@Override
+	public boolean hasChildren() throws TskCoreException {
+		return false;
+	}
+
+	@Override
+	public int getChildrenCount() throws TskCoreException {
+		return 0;
+	}
+
+	@Override
+	public Content getParent() throws TskCoreException {
+		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
+	}
+
+	@Override
+	public List<Long> getChildrenIds() throws TskCoreException {
+		return Collections.<Long>emptyList();
+	}
+
+	@Override
+	public BlackboardArtifact newArtifact(int artifactTypeID) throws TskCoreException {
+		if (artifactTypeID != BlackboardArtifact.ARTIFACT_TYPE.TSK_KEYWORD_HIT.getTypeID()) {
+			throw new TskCoreException("Reports can only have keyword hit artifacts.");
+		}
+		return db.newBlackboardArtifact(artifactTypeID, id);
+	}
+
+	@Override
+	public BlackboardArtifact newArtifact(BlackboardArtifact.ARTIFACT_TYPE type) throws TskCoreException {
+		return newArtifact(type.getTypeID());
+	}
+
+	@Override
+	public ArrayList<BlackboardArtifact> getArtifacts(String artifactTypeName) throws TskCoreException {
+		return getArtifacts(db.getArtifactType(artifactTypeName).getTypeID());
+	}
+
+	@Override
+	public BlackboardArtifact getGenInfoArtifact() throws TskCoreException {
+		// TSK_GEN_INFO artifact is obsolete.
+		return null;
+	}
+
+	@Override
+	public BlackboardArtifact getGenInfoArtifact(boolean create) throws TskCoreException {
+		// TSK_GEN_INFO artifact is obsolete.
+		return null;
+	}
+
+	@Override
+	public ArrayList<BlackboardAttribute> getGenInfoAttributes(BlackboardAttribute.ATTRIBUTE_TYPE attr_type) throws TskCoreException {
+		// TSK_GEN_INFO artifact is obsolete.
+		return null;
+	}
+
+	@Override
+	public ArrayList<BlackboardArtifact> getArtifacts(int artifactTypeID) throws TskCoreException {
+		if (artifactTypeID != BlackboardArtifact.ARTIFACT_TYPE.TSK_KEYWORD_HIT.getTypeID()) {
+			throw new TskCoreException("Reports can only have keyword hit artifacts.");
+		}
+		return db.getBlackboardArtifacts(artifactTypeID, id);
+	}
+
+	@Override
+	public ArrayList<BlackboardArtifact> getArtifacts(BlackboardArtifact.ARTIFACT_TYPE type) throws TskCoreException {
+		return getArtifacts(type.getTypeID());
+	}
+
+	@Override
+	public ArrayList<BlackboardArtifact> getAllArtifacts() throws TskCoreException {
+		return db.getMatchingArtifacts("WHERE obj_id = " + id); //NON-NLS
+	}
+
+	@Override
+	public Set<String> getHashSetNames() throws TskCoreException {
+		return Collections.<String>emptySet();
+	}
+
+	@Override
+	public long getArtifactsCount(String artifactTypeName) throws TskCoreException {
+		return db.getBlackboardArtifactsCount(artifactTypeName, id);
+	}
+
+	@Override
+	public long getArtifactsCount(int artifactTypeID) throws TskCoreException {
+		if (artifactTypeID != BlackboardArtifact.ARTIFACT_TYPE.TSK_KEYWORD_HIT.getTypeID()) {
+			throw new TskCoreException("Reports can only have keyword hit artifacts.");
+		}
```

Comment:
Why throw for this getArtifactsCount overload, but not for the others? It seems to me that we should be consistent.

Reply:
I thought I had covered all the others...where is it missing. Note that all the others end up calling this method.

Reply:
I figured that was your intent, but it looks like 

public long getArtifactsCount(String artifactTypeName) throws TskCoreException

goes straight to the database without calling this overload.

Reply:
Ok, looks like I missed that one. I'll take care of this in a subsequent pull request.